### DISCUSSION TITLE
Git checkout commit -- file after a merge

### DISCUSSION THREAD

**Discussion by pete4abw** [community/community#22497] [New to GitHub]
<p>I performed a merge which went fine</p>
<pre><code class="lang-auto">git checkout branchA
git merge branchB
</code></pre>
<p>The README file however was unique to <code>branchB</code>, so I wanted to replace it with <code>branchA</code>'s README. So, I performed</p>
<p><code>git checkout commit -- README</code>  (where <em>commit</em> was just prior to the merge commit where I last updated the branchA README)</p>
<p>to restore the file. And README was restored. The HEAD is not detached. Exactly what I wanted.</p>
<p>My questions are:</p>
<ol>
<li>Why does <code>git status</code> not show any uncommitted changes?</li>
<li>When I perform a <code>git push</code>, will the restored README be pushed along with all the other merges?</li>
<li>Is there a record somewhere where the checkout overwrote the merged version of README?</li>
</ol>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="pete4abw" data-post="1" data-topic="231098">
<div class="title">
<div class="quote-controls"></div>
 pete4abw:</div>
<blockquote>
<p>Why does <code>git status</code> not show any uncommitted changes?</p>
</blockquote>
</aside>
<p>Most likely that means that the README after merge was already the same as on branchA. It’s impossible to say for sure without seeing your <code>git status</code> output and data, though.</p>
<p>The answers to the other questions depend on this one. :wink:</p>

**Comment by pete4abw** [comment]
<p>branchB README &lt;&gt; branchA README<br>
After merge, branchA README == branchB README</p>
<p><code>git status</code> showed no changes.</p>
<p>branchB</p>
<pre><code class="lang-auto">$ git log README.md
 b7636d5 2022-02-10  (tag: v0.8.9) Documentation Updates, VERSION BUMP
 e79df85 2021-12-02  (tag: v0.8.8) VERSION BUMP. Update Docs.
</code></pre>
<p>branchA <strong>after</strong> merge and <strong>after</strong></p>
<pre><code class="lang-auto">$ git merge branchB
$ git checkout e79df85 -- README.md
$ git log README.md
 e79df85 2021-12-02  (tag: v0.8.8) VERSION BUMP. Update Docs.
 54fa46d 2021-11-29  Announce current branch, lzma-21.06,  is default.
 fda3563 2021-11-28  (tag: v0.8.7) VERSION BUMP. Version Doc Update.
</code></pre>
<p>Almost like the merge of that file never happened! YET the branchB change is reflected in branchA</p>
<pre><code class="lang-auto">$ git log -p b7636d5
 b7636d5 2022-02-10  (tag: v0.8.9) Documentation Updates, VERSION BUMP

diff --git a/README.md b/README.md
index 2fe5b9d..7bf3a5c 100644
--- a/README.md
+++ b/README.md
@@ -1,204 +1,15 @@
 lrzip-next - Long Range ZIP
 ======================
 changes.....
</code></pre>
<p>Additional thoughts?</p>

**Comment by pete4abw** [comment, marked as answer]
<p>This is on me.</p>
<p>If I had just done an additional <code>git status</code> I would have seen that by restoring the file from the index, it became staged and ready for committing. Of course, this is expected behavior. Thank you for your input <a class="mention" href="/airtower-luna">@airtower-luna</a> .</p>
<pre><code class="lang-auto">$ git ls-files -s README.md
100644 7bf3a5ce09c2059d68470f5e069b9f28f87727b4 0       README.md
$ git checkout e79df85 -- README.md
$ git ls-files -s README.md
100644 2fe5b9d1a799e2c8e5ce3f3b7c50deed9ce23e03 0       README.md
$ git status
On branch lzma-21.06
Your branch is ahead of 'origin/lzma-21.06' by 8 commits.
  (use "git push" to publish your local commits)

Changes to be committed:
  (use "git restore --staged &lt;file&gt;..." to unstage)
        modified:   README.md
</code></pre>
